### PR TITLE
options/posix: Wire up some functions to syscalls

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR wires the following functions up to a syscall (syscall is unimplemented for now, as this requires work in managarm first).

- `sync()`, wired up to a `sys_sync()` function,
- `fdatasync(int fd)`, wired up to a `sys_fdatasync(int fd)` function.

This PR technically fulfills #51 and #55, but seeing that they are not implemented in any sysdep (and certainly not in managarm, as the issue states), they, just like #168, can **not** be closed by this issue.

### CODE REVIEW

File: options/posix/generic/unistd-stubs.cpp [hunk 1]
```diff
@@ -600,9 +612,16 @@ int symlinkat(const char *target_path, int dirfd, const char *link_path) {
 	}
 	return 0;
 }
+
 void sync(void) {
-	__ensure(!"Not implemented");
+	if(!mlibc::sys_sync) {
+		MLIBC_MISSING_SYSDEP();
+		errno = ENOSYS;
```
Comment (avdgrinten) [inline options/posix/generic/unistd-stubs.cpp:619]: `sync()` should not set errno.

**Review by avdgrinten — COMMENTED** [review]
Small comment below. I think we can actually close these mlibc issues even though no sysdeps implement the hooks. From mlibc's side, the work is done here; now the sysdeps need to deliver implementatios.

**Comment by Dennisbonke** [comment]
Considering that the issues explicitly state that on managarm, that function is unimplemented, I kinda see that as please implement the sysdep, then close the issue, but that is my opinion

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**